opennms-doc/guide-development/src/asciidoc/text/graph/graph.adoc

```diff
@@ -0,0 +1,201 @@
+// Allow image rendering
+:imagesdir: ../../images
+
+This section covers the insides of the new _Graph Service API_.
+The _Graph Service API_ was heavily inspired by the link:http://graphml.graphdrawing.org[GraphML] model and the original Topology API.
+
+
+[[ga-development-graph-service-api-terminology]]
+=== Terminology
+
+A *Graph* consists of any number of points with any number of connections in between.
+Usually these points are called *Nodes*, but as *Nodes* have a specific meaning in _{opennms-product-name}_ in the context of the _Graph Service_ they are called *Vertices* (plural of *Vertex*).
+The connection between two *Vertices* are usually called *Link*, however it also has a specific meaning inside _{opennms-product-name}_, thus it is called *Edge*.
+A *Graph Container* contains any number of *Graphs*.
+In general an *Element* can either be a *Graph*, *Vertex* or *Edge*.
+
+
+=== Graph Model
+
+[[ga-development-graph-service-api-generic-model]]
+==== Generic Graph Model
+
+The _Generic Graph Model_ is very similar to link:http://graphml.graphdrawing.org[GraphML]'s model and consists of the basic elements:
+
+* `GenericGraphContainer`,
+* `GenericGraph`,
+* `GenericVertex` and
+* `GenericEdge`
+
+Each of the above have besides their designated functionality described in link:#ga-development-graph-service-api-terminology[Terminology] also properties assigned to them.
+Some properties are mandatory and determined by the element's type.
+Others are optionally and totally up to the entity providing a graph to the system.
+
+The following rules apply to the _Generic Graph Model_:
+
+- Each Graph container must be uniquely identified.
```

Comment:
somehow we have the explanation about identifiers twice. Here and in chapter Identifier and Namespace. Maybe consolidate?

Reply:
DONE